lib/lib_common.h

```diff
@@ -0,0 +1,18 @@
+#ifndef SF_LIB_COMMON_H
+#define SF_LIB_COMMON_H
+
+typedef enum SF_RET_CODE
```

Comment:
Can you merge this file with `client_int.h`

Reply:
Also can you rename this to SF_INT*. Just so we know these are for internal use only.

Reply:
I had thought about putting this in client_int.h too but lib_common is included by both tree_map and rbtree both of which do not need the to know the information present in client_int.h and hence I made this. The file might look small now but the goal is to update the all libs/ to use these return types. This might warrant for a *_common.h for the lib folder.

Reply:
If the idea is that the whole library will use those return types internally, shouldn't we put it in `client_int.h` since the whole library will use those values? `client_int.h` is used to store code that is used across all of the lib files which would work well for the return types. This file is only 100 lines long and even if we end up having 100s of return types, thats still a small file. In the future we can split it up once this file gets larger, but right now I think its a bit overkill.

Reply:
cool

Reply:
On second thoughts I think it makes sense for now to have a separate file because including client_int.h as it is currently, exposes a lot of un-necessary code to unrelated files. For instance client_int.h includes client.h the contents of which do not make sense to be exposed to the data structure files like rbtree / treemap / paramstore etc. Essentially the main point here is code separation.